File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] words) {
       


        HashMap<Character,Integer> mp = new HashMap<>();
        int n  = words.length;
        for(String word: words){
            for(Character c : word.toCharArray())
               mp.put(c, mp.getOrDefault(c,0)+1 );
        }
        for(Integer count: mp.values())
            if(count%n!=0)
                return false;
        return true ;
    }
}

File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.py
from collections import defaultdict as dict 

class Solution:
    def makeEqual(self, words: List[str]) -> bool:

        mp =dict(int)
        n  = len(words)
        for word  in words:
            for c in word:

                mp[c]+=1 
    
        for c in mp:
            if mp[c]%n!=0:
                return False
        return True